server/src/main/java/org/opensearch/indices/replication/SegmentReplicationState.java

```diff
@@ -60,23 +66,42 @@ public static Stage fromId(byte id) {
 
     private Stage stage;
     private final ReplicationLuceneIndex index;
-    private final ReplicationTimer timer;
+    private final ReplicationTimer overallTimer;
+    private final ReplicationTimer stepTimer;
+    private final List<Tuple<String, Long>> timingData;
+    private long replicationId;
 
     public SegmentReplicationState(ReplicationLuceneIndex index) {
         stage = Stage.INIT;
         this.index = index;
-        timer = new ReplicationTimer();
-        timer.start();
+        // Timing data will have as many entries as stages, plus one
+        // additional entry for the overall timer
+        timingData = new ArrayList<>(Stage.values().length + 1);
+        overallTimer = new ReplicationTimer();
+        stepTimer = new ReplicationTimer();
+        stepTimer.start();
     }
 
     @Override
     public ReplicationLuceneIndex getIndex() {
         return index;
     }
 
+    public long getReplicationId() {
+        return replicationId;
+    }
+
+    public void setReplicationId(long replicationId) {
```

Comment:
Under which circustances the replication id could be changed for existing state?

Reply:
@reta Fair point. There are no situations where the replication ID will change for an existing seg-rep state object. Instead of this method, I'll introduce an overloaded constructor instead.

Reply:
:+1: